subsys/bluetooth/services/fast_pair/fp_registration_data.c

```diff
@@ -24,13 +25,14 @@ static const uint8_t fp_magic[] = {0xFA, 0x57, 0xFA, 0x57};
 #define FP_OFFSET(_prev_item_offset, _prev_item_len) \
 	(_prev_item_offset + ROUND_UP(_prev_item_len, FP_DATA_ALIGN))
 
-#define FP_MAGIC_HEADER_OFF	 FP_DATA_START
-#define FP_MODEL_ID_OFF		 FP_OFFSET(FP_MAGIC_HEADER_OFF, FP_MAGIC_SIZE)
+#define FP_MAGIC_OFF		 FP_DATA_START
+#define FP_MODEL_ID_OFF		 FP_OFFSET(FP_MAGIC_OFF, FP_MAGIC_SIZE)
 #define FP_ANTI_SPOOFING_KEY_OFF FP_OFFSET(FP_MODEL_ID_OFF, FP_REG_DATA_MODEL_ID_LEN)
-#define FP_MAGIC_TRAILER_OFF	 FP_OFFSET(FP_ANTI_SPOOFING_KEY_OFF, \
+#define FP_HASH_OFF		 FP_OFFSET(FP_ANTI_SPOOFING_KEY_OFF, \
 					   FP_REG_DATA_ANTI_SPOOFING_PRIV_KEY_LEN)
+#define FP_PROV_DATA_SIZE	 (FP_HASH_OFF - FP_DATA_START)
```

Comment:
The `FP_PROV_DATA_SIZE` name suggests that it relates to whole provisioning data which could be a bit confusing (we take magic into account, but we do not take sha256 into account)

Reply:
Maybe we could take hash into account here too? Then we could read whole provisioning at once from FLASH to a buffer and then assign pointers to places denoted by the offsets. The verification functions could still be separated for readability.